OMCompiler/SimulationRuntime/c/simulation/solver/spatialDistribution.c

```diff
@@ -272,6 +274,22 @@ void storeSpatialDistribution(DATA* data, threadData_t *threadData, unsigned int
     omc_throw_function(threadData);
   }
 
+  /* Get deltaX */
+  deltaX = fabs(spatialDistribution->oldPosX - posX);
+  if (posX - spatialDistribution->oldPosX > 0) {
+    realDirection = 1 /* positive */;
+  } else if (posX - spatialDistribution->oldPosX < 0) {
+    realDirection = -1 /* negative */;
+  } else {
+    realDirection = 0 /* standing still */;
+  }
+
+  /* Check if real dirrection matches isPositiveVelocity */
+  if ((isPositiveVelocity && realDirection==-1 && deltaX > SPATIAL_ZERO_DELTA_X) || (!isPositiveVelocity && realDirection!= -1 && deltaX > SPATIAL_ZERO_DELTA_X)) {
```

Comment:
I don't really understand this check. As far as i understand both `isPositiveVelocity ` and `realDirection` can be -1, 0 or 1.

If you just check if the value itself as condition, it just checks if it is nonzero right? Why do you not check for `isPositiveVelocity == realDirection && deltaX > SPATIAL_ZERO_DELTA_X`? Is it correct that you allow the combinations
(0,1) and (0,-1) to return true? Also shouldn't it also return true if `deltaX < SPATIAL_ZERO_DELTA_X`?

Reply:
If any of them can be zero and it is supposed to return true i would simply do `!((isPositiveVelocity * realDirection) < 0) `

Reply:
The condition should hold true if `isPositiveVelocity` doesn't match `realDirection`. If `realDirection=0` we don't care

So yes, it would be the same to do
```C
  if (deltaX > SPATIAL_ZERO_DELTA_X && isPositiveVelocity*realDirection > 0 ) {
    isPositiveVelocity = isPositiveVelocity ? 0:1;
  }
```
but is this more easy to understand? Anyway I'll add it since it is shorter.